diff --git a/src/GameApplication.php b/src/GameApplication.php
index 0cb97a2..dc726b6 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -2,6 +2,7 @@
 
 namespace App;
 
+use App\ActionCommand\AttackCommand;
 use App\Builder\CharacterBuilder;
 use App\Character\Character;
 use App\Observer\GameObserverInterface;
@@ -31,18 +32,8 @@ class GameApplication
             ), '']);
 
             // Player's turn
-            $playerDamage = $player->attack();
-            if ($playerDamage === 0) {
-                GameApplication::$printer->printFor($player)->exhaustedMessage();
-                $fightResultSet->of($player)->addExhaustedTurn();
-            }
-
-            $damageDealt = $ai->receiveAttack($playerDamage);
-            $fightResultSet->of($player)->addDamageDealt($damageDealt);
-
-            GameApplication::$printer->printFor($player)->attackMessage($damageDealt);
-            GameApplication::$printer->writeln('');
-            usleep(300000);
+            $attackCommand = new AttackCommand($player, $ai, $fightResultSet);
+            $attackCommand->execute();
 
             if ($this->didPlayerDie($ai)) {
                 $this->endBattle($fightResultSet, $player, $ai);
@@ -50,18 +41,8 @@ class GameApplication
             }
 
             // AI's turn
-            $aiDamage = $ai->attack();
-
-            if ($aiDamage === 0) {
-                GameApplication::$printer->printFor($ai)->exhaustedMessage();
-                $fightResultSet->of($ai)->addExhaustedTurn();
-            }
-
-            $damageReceived = $player->receiveAttack($aiDamage);
-            $fightResultSet->of($player)->addDamageReceived($damageReceived);
-
-            GameApplication::$printer->printFor($ai)->attackMessage($damageReceived);
-            GameApplication::$printer->writeln('');
+            $aiAttackCommand = new AttackCommand($ai, $player, $fightResultSet);
+            $aiAttackCommand->execute();
 
             if ($this->didPlayerDie($player)) {
                 $this->endBattle($fightResultSet, $ai, $player);
