diff --git a/src/GameApplication.php b/src/GameApplication.php
index 77d83c8..3aae9c9 100644
--- a/src/GameApplication.php
+++ b/src/GameApplication.php
@@ -110,6 +110,9 @@ class GameApplication
         $fightResultSet->of($winner)->addVictory();
         $fightResultSet->of($loser)->addDefeat();
 
+        $xpBonus = $this->xpBonusHandler->handle($winner, $fightResultSet->of($winner));
+        $winner->addXp($xpBonus);
+
         $this->notify($fightResultSet);
 
         $winner->rest();
