diff --git a/src/ChainHandler/CasinoHandler.php b/src/ChainHandler/CasinoHandler.php
index 8302aff..dd15ac4 100644
--- a/src/ChainHandler/CasinoHandler.php
+++ b/src/ChainHandler/CasinoHandler.php
@@ -11,6 +11,11 @@ class CasinoHandler implements XpBonusHandlerInterface
 {
     private XpBonusHandlerInterface $next;
 
+    public function __construct()
+    {
+        $this->next = new NullHandler();
+    }
+
     public function handle(Character $player, FightResult $fightResult): int
     {
         $dice1 = Dice::roll(6);
@@ -26,11 +31,7 @@ class CasinoHandler implements XpBonusHandlerInterface
             return 25;
         }
 
-        if (isset($this->next)) {
-            return $this->next->handle($player, $fightResult);
-        }
-
-        return 0;
+        return $this->next->handle($player, $fightResult);
     }
 
     public function setNext(XpBonusHandlerInterface $next): void
diff --git a/src/ChainHandler/LevelHandler.php b/src/ChainHandler/LevelHandler.php
index 4857957..d482f35 100644
--- a/src/ChainHandler/LevelHandler.php
+++ b/src/ChainHandler/LevelHandler.php
@@ -10,17 +10,18 @@ class LevelHandler implements XpBonusHandlerInterface
 {
     private XpBonusHandlerInterface $next;
 
+    public function __construct()
+    {
+        $this->next = new NullHandler();
+    }
+
     public function handle(Character $player, FightResult $fightResult): int
     {
         if ($player->getLevel() === 1) {
             return 25;
         }
 
-        if (isset($this->next)) {
-            return $this->next->handle($player, $fightResult);
-        }
-
-        return 0;
+        return $this->next->handle($player, $fightResult);
     }
 
     public function setNext(XpBonusHandlerInterface $next): void
diff --git a/src/ChainHandler/OnFireHandler.php b/src/ChainHandler/OnFireHandler.php
index 8e838bd..672cef8 100644
--- a/src/ChainHandler/OnFireHandler.php
+++ b/src/ChainHandler/OnFireHandler.php
@@ -10,17 +10,18 @@ class OnFireHandler implements XpBonusHandlerInterface
 {
     private XpBonusHandlerInterface $next;
 
+    public function __construct()
+    {
+        $this->next = new NullHandler();
+    }
+
     public function handle(Character $player, FightResult $fightResult): int
     {
         if ($fightResult->getWinStreak() >= 3) {
             return 25;
         }
 
-        if (isset($this->next)) {
-            return $this->next->handle($player, $fightResult);
-        }
-
-        return 0;
+        return $this->next->handle($player, $fightResult);
     }
 
     public function setNext(XpBonusHandlerInterface $next): void
